src/app/component/key/typer-interface.js

```diff
@@ -0,0 +1,98 @@
+class TyperInterface {
+  constructor() {
+    this.target = null;
+    this.selection = {
```

Comment:
If you class properties doesn't depend on pass argument, you can set them above the constructor.